### PR TITLE
VertexAI: migrate test cases to use updated set of golden files 

### PR DESCRIPTION
**Pull request by tanzimfh**
This updates VertexAI unit tests to work with the golden files that will be shared with Android and JS.

### CODE REVIEW

File: FirebaseVertexAI/Tests/Unit/Responses/unary-success-basic-reply-short.json [hunk 1]
```diff
@@ -47,4 +48,4 @@
     "candidatesTokenCount": 7,
     "totalTokenCount": 13
   }
-}
+}
```
Comment (andrewheard) [inline FirebaseVertexAI/Tests/Unit/Responses/unary-success-basic-reply-short.json:51]: Would it be alright if we added the trailing newline back in this one? It doesn't affect the unit tests but I double checked with `curl ... -o filename.json` and it seems like the response does include one. Xcode also adds one when I open the file.

File: FirebaseVertexAI/Tests/Unit/Responses/streaming-failure-invalid-json.txt
Comment (andrewheard) [inline FirebaseVertexAI/Tests/Unit/Responses/streaming-failure-invalid-json.txt:1]: Would you mind adding a second blank line to the end of this file while you're moving it? I just realized I forgot to add one.

File: FirebaseVertexAI/Tests/Unit/Responses/streaming-failure-malformed-content.txt
Comment (andrewheard) [inline FirebaseVertexAI/Tests/Unit/Responses/streaming-failure-malformed-content.txt:1]: Please add a second trailing newline here too.

File: FirebaseVertexAI/Tests/Unit/Responses/streaming-failure-unknown-finish-enum.txt
Comment (andrewheard) [inline FirebaseVertexAI/Tests/Unit/Responses/streaming-failure-unknown-finish-enum.txt:1]: Please add a second trailing newline here too (sorry I think that's all of them now).

**Review by andrewheard — APPROVED** [review]
Thanks, Tanzim! This LGTM!

File: FirebaseVertexAI/Tests/Unit/ChatTests.swift [hunk 2]
```diff
@@ -69,7 +69,8 @@ final class ChatTests: XCTestCase {
...
...hey use for communication and expressing emotions.\n\n- **Behavior and Personality:**\n  - Loyal: Dogs are known for their loyalty and devotion to their owners.\n  - Friendly: Dogs are generally friendly and outgoing animals that enjoy interacting with people and other animals.\n  - Playful: Dogs are playful and energetic creatures that love to engage in activities such as fetching, running, and playing with toys.\n  - Trainable: Dogs are highly trainable and can learn a variety of commands and tricks.\n  - Vocal: Dogs communicate through a variety of vocalizations, including barking, howling, whining, and growling.\n\n- **Health and Care:**\n  - Diet: Dogs are omnivores and can eat a variety of foods, including meat, vegetables, and grains.\n  - Grooming: Dogs require regular grooming to keep their fur clean and free of mats. The frequency of grooming depends on the breed and coat type.\n  - Exercise: Dogs need regular exercise to stay healthy and active. The amount of exercise required varies depending on the breed and age of the dog.\n  - Veterinary Care: Regular veterinary checkups are essential for maintaining a dog's health and detecting any potential health issues early on."
```
Comment (rlazo) [inline FirebaseVertexAI/Tests/Unit/ChatTests.swift:73]: @andrewheard is this something that can be partially matched? I'm not sure if matching the whole text is the most maintainable scenario.

Comment (andrewheard) [reply FirebaseVertexAI/Tests/Unit/ChatTests.swift:73]: I think the intention here was specifically to test that the whole text does match based on the test name (`testMergingText`). I wonder if we could make a new `.txt` for this test and revert back to the `1 2 3 4 5 6 7 8`. I see that it was originally a long response and Ryan shortened it in https://github.com/google-gemini/generative-ai-swift/pull/50.

Comment (tanzimfh) [reply FirebaseVertexAI/Tests/Unit/ChatTests.swift:73]: That SGTM. Should I add back Ryan's file as "streaming-success-basic-reply-parts.txt" and revert the test to "1 2 3 4 5 6 7 8"?

Comment (andrewheard) [reply FirebaseVertexAI/Tests/Unit/ChatTests.swift]: SGTM. Thanks!

**Review by andrewheard — APPROVED** [review]

**Review by rlazo — APPROVED** [review]

**Pull request merged**